src/utils/objects-have-same-keys.js

```diff
@@ -0,0 +1,16 @@
+/**
+   * Compare two sets of objects keys, returns true if both have the same keys.
+   * An error has to be thrown in HookLoad.getResponse() when a field is deleted or added
+   * (new key or missing key in fields).
+   * Source: https://stackoverflow.com/a/35047888/978690
+   *
+   * @param  {...any} objects The list of fields object we want to compare
+   * @see HookLoad.getResponse() for usage
+ */
+function objectsHaveSameKeys(...objects) {
+  const allKeys = objects.reduce((keys, object) => keys.concat(Object.keys(object)), []);
+  const union = new Set(allKeys);
+  return objects.every((object) => union.size === Object.keys(object).length);
+}
+
+module.exports = objectsHaveSameKeys;
```

Comment:
We can even go further and simplify with:
```javascript
const isSameDataStructure = (fields, previousFields, deep = 0, step = 0) => {
    if (!fields
        || !previousFields
        || !(typeof fields === 'object' && typeof previousFields === 'object')) {
        return false;
    }

    const fieldsKeys = Object.keys(fields); // O(n)
    const previousFieldsKeys = Object.keys(previousFields);  // O(m)
    if (fieldsKeys.length !== previousFieldsKeys.length) {
        return false;
    }

    // NOTICE: use Set because `has` method has a time complexity of `O(1)`.
    const fieldsKeysSet = new Set(fieldsKeys);  // O(n)
    for (const key of previousFieldsKeys) {  // O(m)
        // NOTICE: if key does not exist in other object or by its children.
        if (!fieldsKeysSet.has(key)
            || (step + 1 <= deep   // recursive, limit to deep
                && !isSameDataStructure(fields[key], previousFields[key], deep, step + 1))) {
            return false;
        }
    }

    return true;
}
```

Reply:
Let's go with this.